### PR TITLE
[AMORO-2993] Enrich optimizing metrics

### LINKED ISSUE #2993: [Improvement]: Enrich table optimizing metrics
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Collect more metrics while self-optimizing

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2993.

## Brief change log

- As mentioned in the issue

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by baiyangtx — COMMENTED** [review]
The new metrics should be added to docs.

https://github.com/apache/amoro/blob/master/docs/user-guides/metrics.md

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOptimizingMetrics.java [hunk 1]
```diff
@@ -163,6 +165,37 @@ public class TableOptimizingMetrics {
           .withTags("catalog", "database", "table")
           .build();
 
+  public static final MetricDefine TABLE_LAST_MINOR_OPTIMIZING_INTERVAL =
+      defineGauge("table_last_minor_optimizing_duration_mills")
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOptimizingMetrics.java:169]: Considering that the metrics related to optimizing on the table are currently prefixed with `table_optimizing`, I suggest renaming this monitoring metric to `table_optimizing_minor_time_mills_since_last_completion` or `table_optimizing_minor_duration_mills_since_last_completion`.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOptimizingMetrics.java [hunk 2]
```diff
@@ -163,6 +165,37 @@ public class TableOptimizingMetrics {
...
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_LAST_OPTIMIZING_DURATION =
+      defineGauge("table_last_optimizing_duration_mills")
+          .withDescription("Duration in milliseconds after last optimizing")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_OPTIMIZING_REFRESHED_SNAPSHOT_SLIDE_INTERVAL =
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOptimizingMetrics.java:192]: Is `table_optimizing_lag_duration_mills` a proper name?
And we should calculate it with: latestSnapshot.time - optimizedSnapshot.time.

**Review by zhoujinsong — COMMENTED** [review]
I left some comments, PTAL.

**Comment by XBaith** [comment]
Hi @baiyangtx, @zhoujinsong update code and add document, PTAL

**Review by baiyangtx — APPROVED** [review]
LGTM

**Comment by XBaith** [comment]
The last refresh snapshot should not be any of optimizing or maintainenace snapshot committed by amoro. So I suggest merging pr #3058 before pushing it forward

**Comment by zhoujinsong** [comment]
@XBaith Some conflicts need to be fixed.

**Comment by XBaith** [comment]
@zhoujinsong @baiyangtx I exclude the snapsthos produced by Amoro, PTAL and feel free to merge this PR

**Review by czy006 — APPROVED** [review]

**Pull request merged**